rmw_implementation/src/functions.cpp

```diff
@@ -607,6 +625,10 @@ void prefetch_symbols(void)
   GET_SYMBOL(rmw_compare_gids_equal)
   GET_SYMBOL(rmw_service_server_is_available)
   GET_SYMBOL(rmw_set_log_severity)
+  GET_SYMBOL(rmw_get_publishers_info_by_topic)
+  rmw_reset_error();
```

Comment:
This is because if the said function is not implemented by the implementation, then the GET_SYMBOL function replaces the previously set error and this prints warnings since rcutils_reset_error is not called.

Reply:
It's true that if more than one symbol is not found, a warning will be printed telling you that the error message was overridden. But again, it's out of the scope of the PR.
Please, open a follow-up issue, and we can apply the result of the discussion to the whole code consistently.